// ionic-angular/components/picker/picker-component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/common';
import * as i2 from '../button/button.ngfactory';
import * as i3 from 'ionic-angular/components/button/button';
import * as i4 from 'ionic-angular/config/config';
import * as i5 from './picker-column.ngfactory';
import * as i6 from 'ionic-angular/components/picker/picker-column';
import * as i7 from 'ionic-angular/platform/platform';
import * as i8 from 'ionic-angular/tap-click/haptic';
import * as i9 from 'ionic-angular/platform/dom-controller';
import * as i10 from 'ionic-angular/components/picker/picker-component';
import * as i11 from 'ionic-angular/components/backdrop/backdrop';
import * as i12 from 'ionic-angular/navigation/view-controller';
import * as i13 from 'ionic-angular/gestures/gesture-controller';
import * as i14 from 'ionic-angular/navigation/nav-params';
const styles_PickerCmp:any[] = ([] as any[]);
export const RenderType_PickerCmp:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_PickerCmp,
    data:{}});
function View_PickerCmp_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),7,'div',[['class',
      'picker-toolbar-button']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),i0.ɵdid(278528,(null as any),0,i1.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
      'ngClass']},(null as any)),(_l()(),i0.ɵted((null as any),['\n          '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),3,'button',[['class','picker-button'],
          ['clear',''],['ion-button','']],(null as any),[[(null as any),'click']],
          (_v,en,$event) => {
            var ad:boolean = true;
            var _co:any = _v.component;
            if (('click' === en)) {
              const pd_0:any = ((<any>_co.btnClick(_v.context.$implicit)) !== false);
              ad = (pd_0 && ad);
            }
            return ad;
          },i2.View_Button_0,i2.RenderType_Button)),i0.ɵdid(278528,(null as any),0,
          i1.NgClass,[i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],
          {klass:[0,'klass'],ngClass:[1,'ngClass']},(null as any)),i0.ɵdid(1097728,
          (null as any),0,i3.Button,[[8,''],i4.Config,i0.ElementRef,i0.Renderer],{clear:[0,
              'clear']},(null as any)),(_l()(),i0.ɵted(0,['\n            ','\n          '])),
      (_l()(),i0.ɵted((null as any),['\n        ']))],(_ck,_v) => {
    const currVal_0:any = 'picker-toolbar-button';
    const currVal_1:any = _v.context.$implicit.cssRole;
    _ck(_v,1,0,currVal_0,currVal_1);
    const currVal_2:any = 'picker-button';
    const currVal_3:any = _v.context.$implicit.cssClass;
    _ck(_v,4,0,currVal_2,currVal_3);
    const currVal_4:any = '';
    _ck(_v,5,0,currVal_4);
  },(_ck,_v) => {
    const currVal_5:any = _v.context.$implicit.text;
    _ck(_v,6,0,currVal_5);
  });
}
function View_PickerCmp_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'div',[['class',
      'picker-col']],[[4,'max-width',(null as any)],[2,'picker-opts-left',(null as any)],
      [2,'picker-opts-right',(null as any)]],[[(null as any),'ionChange']],(_v,en,
      $event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('ionChange' === en)) {
      const pd_0:any = ((<any>_co._colChange($event)) !== false);
      ad = (pd_0 && ad);
    }
    return ad;
  },i5.View_PickerColumnCmp_0,i5.RenderType_PickerColumnCmp)),i0.ɵdid(4374528,[[1,
      4]],0,i6.PickerColumnCmp,[i4.Config,i7.Platform,i0.ElementRef,i0.NgZone,i8.Haptic,
      i7.Platform,i9.DomController],{col:[0,'col']},{ionChange:'ionChange'})],(_ck,
      _v) => {
    const currVal_3:any = _v.context.$implicit;
    _ck(_v,1,0,currVal_3);
  },(_ck,_v) => {
    const currVal_0:any = i0.ɵnov(_v,1).col.columnWidth;
    const currVal_1:any = (i0.ɵnov(_v,1).col.align == 'left');
    const currVal_2:any = (i0.ɵnov(_v,1).col.align == 'right');
    _ck(_v,0,0,currVal_0,currVal_1,currVal_2);
  });
}
export function View_PickerCmp_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[i0.ɵqud(671088640,1,{_cols:1}),(_l()(),i0.ɵted((null as any),['\n    '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),1,'ion-backdrop',[['disable-activated',
          ''],['role','presentation'],['tappable','']],(null as any),[[(null as any),
          'click']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:i10.PickerCmp = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_co.bdClick()) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),i0.ɵdid(16384,(null as any),0,i11.Backdrop,[i0.ElementRef,
          i0.Renderer],(null as any),(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n    '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),17,'div',[['class',
          'picker-wrapper']],(null as any),(null as any),(null as any),(null as any),
          (null as any))),(_l()(),i0.ɵted((null as any),['\n      '])),(_l()(),i0.ɵeld(0,
          (null as any),(null as any),4,'div',[['class','picker-toolbar']],(null as any),
          (null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['\n        '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,
          (null as any),View_PickerCmp_1)),i0.ɵdid(802816,(null as any),0,i1.NgForOf,
          [i0.ViewContainerRef,i0.TemplateRef,i0.IterableDiffers],{ngForOf:[0,'ngForOf']},
          (null as any)),(_l()(),i0.ɵted((null as any),['\n      '])),(_l()(),i0.ɵted((null as any),
          ['\n      '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),8,'div',[['class',
          'picker-columns']],(null as any),(null as any),(null as any),(null as any),
          (null as any))),(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵeld(0,
          (null as any),(null as any),0,'div',[['class','picker-above-highlight']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵand(16777216,(null as any),
          (null as any),1,(null as any),View_PickerCmp_2)),i0.ɵdid(802816,(null as any),
          0,i1.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,i0.IterableDiffers],{ngForOf:[0,
              'ngForOf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n        '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),0,'div',[['class','picker-below-highlight']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n      '])),(_l()(),i0.ɵted((null as any),['\n    '])),
      (_l()(),i0.ɵted((null as any),['\n  ']))],(_ck,_v) => {
    var _co:i10.PickerCmp = _v.component;
    const currVal_0:any = _co.d.buttons;
    _ck(_v,10,0,currVal_0);
    const currVal_1:any = _co.d.columns;
    _ck(_v,18,0,currVal_1);
  },(null as any));
}
export function View_PickerCmp_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'ion-picker-cmp',
      [['role','dialog']],(null as any),[['body','keyup']],(_v,en,$event) => {
        var ad:boolean = true;
        if (('body:keyup' === en)) {
          const pd_0:any = ((<any>i0.ɵnov(_v,1)._keyUp($event)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },View_PickerCmp_0,RenderType_PickerCmp)),i0.ɵdid(180224,(null as any),0,i10.PickerCmp,
      [i12.ViewController,i0.ElementRef,i4.Config,i13.GestureController,i14.NavParams,
          i0.Renderer],(null as any),(null as any))],(null as any),(null as any));
}
export const PickerCmpNgFactory:i0.ComponentFactory<i10.PickerCmp> = i0.ɵccf('ion-picker-cmp',
    i10.PickerCmp,View_PickerCmp_Host_0,{},{},([] as any[]));
