// ionic-angular/components/picker/picker-column.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/common';
import * as i2 from 'ionic-angular/components/picker/picker-column';
import * as i3 from 'ionic-angular/config/config';
import * as i4 from 'ionic-angular/platform/platform';
import * as i5 from 'ionic-angular/tap-click/haptic';
import * as i6 from 'ionic-angular/platform/dom-controller';
const styles_PickerColumnCmp:any[] = ([] as any[]);
export const RenderType_PickerColumnCmp:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_PickerColumnCmp,data:{}});
function View_PickerColumnCmp_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'div',[['class',
      'picker-prefix']],[[4,'width',(null as any)]],(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵted((null as any),['','']))],(null as any),(_ck,
      _v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.col.prefixWidth;
    _ck(_v,0,0,currVal_0);
    const currVal_1:any = _co.col.prefix;
    _ck(_v,1,0,currVal_1);
  });
}
function View_PickerColumnCmp_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'button',[['class',
      'picker-opt'],['disable-activated','']],[[2,'picker-opt-disabled',(null as any)]],
      [[(null as any),'click']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_co.optClick($event,_v.context.index)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['','']))],(null as any),
      (_ck,_v) => {
        const currVal_0:any = _v.context.$implicit.disabled;
        _ck(_v,0,0,currVal_0);
        const currVal_1:any = _v.context.$implicit.text;
        _ck(_v,1,0,currVal_1);
      });
}
function View_PickerColumnCmp_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'div',[['class',
      'picker-suffix']],[[4,'width',(null as any)]],(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵted((null as any),['','']))],(null as any),(_ck,
      _v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.col.suffixWidth;
    _ck(_v,0,0,currVal_0);
    const currVal_1:any = _co.col.suffix;
    _ck(_v,1,0,currVal_1);
  });
}
export function View_PickerColumnCmp_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[i0.ɵqud(402653184,1,{colEle:0}),(_l()(),i0.ɵand(16777216,(null as any),
      (null as any),1,(null as any),View_PickerColumnCmp_1)),i0.ɵdid(16384,(null as any),
      0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),
      (_l()(),i0.ɵeld(0,[[1,0],['colEle',1]],(null as any),2,'div',[['class','picker-opts']],
          [[4,'max-width',(null as any)]],(null as any),(null as any),(null as any),
          (null as any))),(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),
          View_PickerColumnCmp_2)),i0.ɵdid(802816,(null as any),0,i1.NgForOf,[i0.ViewContainerRef,
          i0.TemplateRef,i0.IterableDiffers],{ngForOf:[0,'ngForOf']},(null as any)),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_PickerColumnCmp_3)),
      i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any))],(_ck,_v) => {
    var _co:i2.PickerColumnCmp = _v.component;
    const currVal_0:any = _co.col.prefix;
    _ck(_v,2,0,currVal_0);
    const currVal_2:any = _co.col.options;
    _ck(_v,5,0,currVal_2);
    const currVal_3:any = _co.col.suffix;
    _ck(_v,7,0,currVal_3);
  },(_ck,_v) => {
    var _co:i2.PickerColumnCmp = _v.component;
    const currVal_1:any = _co.col.optionsWidth;
    _ck(_v,3,0,currVal_1);
  });
}
export function View_PickerColumnCmp_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'div',[['class',
      'picker-col']],[[4,'max-width',(null as any)],[2,'picker-opts-left',(null as any)],
      [2,'picker-opts-right',(null as any)]],(null as any),(null as any),View_PickerColumnCmp_0,
      RenderType_PickerColumnCmp)),i0.ɵdid(4374528,(null as any),0,i2.PickerColumnCmp,
      [i3.Config,i4.Platform,i0.ElementRef,i0.NgZone,i5.Haptic,i4.Platform,i6.DomController],
      (null as any),(null as any))],(null as any),(_ck,_v) => {
    const currVal_0:any = i0.ɵnov(_v,1).col.columnWidth;
    const currVal_1:any = (i0.ɵnov(_v,1).col.align == 'left');
    const currVal_2:any = (i0.ɵnov(_v,1).col.align == 'right');
    _ck(_v,0,0,currVal_0,currVal_1,currVal_2);
  });
}
export const PickerColumnCmpNgFactory:i0.ComponentFactory<i2.PickerColumnCmp> = i0.ɵccf('.picker-col',
    i2.PickerColumnCmp,View_PickerColumnCmp_Host_0,{col:'col'},{ionChange:'ionChange'},
    ([] as any[]));
