// ionic-angular/components/item/item.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from 'ionic-angular/components/label/label';
import * as i2 from 'ionic-angular/config/config';
import * as i3 from './reorder.ngfactory';
import * as i4 from 'ionic-angular/components/item/reorder';
import * as i5 from '@angular/common';
import * as i6 from 'ionic-angular/components/item/item';
import * as i7 from 'ionic-angular/util/form';
import * as i8 from 'ionic-angular/components/item/item-reorder';
const styles_Item:any[] = ([] as any[]);
export const RenderType_Item:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_Item,
    data:{}});
function View_Item_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'ion-label',([] as any[]),
      (null as any),(null as any),(null as any),(null as any),(null as any))),i0.ɵdid(16384,
      [[1,4]],0,i1.Label,[i2.Config,i0.ElementRef,i0.Renderer,[8,(null as any)],[8,
          (null as any)],[8,(null as any)],[8,(null as any)]],(null as any),(null as any)),
      i0.ɵncd((null as any),2)],(null as any),(null as any));
}
function View_Item_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'ion-reorder',
      ([] as any[]),(null as any),[[(null as any),'click']],(_v,en,$event) => {
        var ad:boolean = true;
        if (('click' === en)) {
          const pd_0:any = ((<any>i0.ɵnov(_v,1).onClick($event)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },i3.View_Reorder_0,i3.RenderType_Reorder)),i0.ɵdid(49152,(null as any),0,i4.Reorder,
      [i0.ElementRef],(null as any),(null as any))],(null as any),(null as any));
}
export function View_Item_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[i0.ɵqud(671088640,1,{viewLabel:0}),i0.ɵncd((null as any),0),(_l()(),
      i0.ɵeld(0,(null as any),(null as any),8,'div',[['class','item-inner']],(null as any),
          (null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵeld(0,
      (null as any),(null as any),4,'div',[['class','input-wrapper']],(null as any),
      (null as any),(null as any),(null as any),(null as any))),i0.ɵncd((null as any),
      1),(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_Item_1)),
      i0.ɵdid(16384,(null as any),0,i5.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),i0.ɵncd((null as any),3),i0.ɵncd((null as any),4),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_Item_2)),
      i0.ɵdid(16384,(null as any),0,i5.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵeld(0,(null as any),(null as any),0,
          'div',[['class','button-effect']],(null as any),(null as any),(null as any),
          (null as any),(null as any)))],(_ck,_v) => {
    var _co:i6.Item = _v.component;
    const currVal_0:any = _co._viewLabel;
    _ck(_v,6,0,currVal_0);
    const currVal_1:any = _co._hasReorder;
    _ck(_v,10,0,currVal_1);
  },(null as any));
}
export function View_Item_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),4,'ion-list-header',
      [['class','item']],(null as any),(null as any),(null as any),View_Item_0,RenderType_Item)),
      i0.ɵdid(1097728,(null as any),3,i6.Item,[i7.Form,i2.Config,i0.ElementRef,i0.Renderer,
          [2,i8.ItemReorder]],(null as any),(null as any)),i0.ɵqud(335544320,1,{contentLabel:0}),
      i0.ɵqud(603979776,2,{_buttons:1}),i0.ɵqud(603979776,3,{_icons:1})],(null as any),
      (null as any));
}
export const ItemNgFactory:i0.ComponentFactory<i6.Item> = i0.ɵccf('ion-list-header,ion-item,[ion-item],ion-item-divider',
    i6.Item,View_Item_Host_0,{color:'color',mode:'mode'},{},['[item-start],[item-left],ion-checkbox:not([item-end]):not([item-right])',
        'ion-label','*','ion-select,ion-input,ion-textarea,ion-datetime,ion-range,[item-content]',
        '[item-end],[item-right],ion-radio,ion-toggle']);
