// ionic-angular/components/spinner/spinner.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/common';
import * as i2 from 'ionic-angular/components/spinner/spinner';
import * as i3 from 'ionic-angular/config/config';
const styles_Spinner:any[] = ([] as any[]);
export const RenderType_Spinner:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_Spinner,
    data:{}});
function View_Spinner_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,':svg:svg',[['viewBox',
      '0 0 64 64']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      i0.ɵdid(278528,(null as any),0,i1.NgStyle,[i0.KeyValueDiffers,i0.ElementRef,
          i0.Renderer],{ngStyle:[0,'ngStyle']},(null as any)),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),0,':svg:circle',[['transform','translate(32,32)']],[[1,'r',
              0]],(null as any),(null as any),(null as any),(null as any)))],(_ck,
      _v) => {
    const currVal_0:any = _v.context.$implicit.style;
    _ck(_v,1,0,currVal_0);
  },(_ck,_v) => {
    const currVal_1:any = _v.context.$implicit.r;
    _ck(_v,2,0,currVal_1);
  });
}
function View_Spinner_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,':svg:svg',[['viewBox',
      '0 0 64 64']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      i0.ɵdid(278528,(null as any),0,i1.NgStyle,[i0.KeyValueDiffers,i0.ElementRef,
          i0.Renderer],{ngStyle:[0,'ngStyle']},(null as any)),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),0,':svg:line',[['transform','translate(32,32)']],[[1,'y1',
              0],[1,'y2',0]],(null as any),(null as any),(null as any),(null as any)))],
      (_ck,_v) => {
        const currVal_0:any = _v.context.$implicit.style;
        _ck(_v,1,0,currVal_0);
      },(_ck,_v) => {
        const currVal_1:any = _v.context.$implicit.y1;
        const currVal_2:any = _v.context.$implicit.y2;
        _ck(_v,2,0,currVal_1,currVal_2);
      });
}
export function View_Spinner_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),
      View_Spinner_1)),i0.ɵdid(802816,(null as any),0,i1.NgForOf,[i0.ViewContainerRef,
      i0.TemplateRef,i0.IterableDiffers],{ngForOf:[0,'ngForOf']},(null as any)),(_l()(),
      i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_Spinner_2)),
      i0.ɵdid(802816,(null as any),0,i1.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,
          i0.IterableDiffers],{ngForOf:[0,'ngForOf']},(null as any))],(_ck,_v) => {
    var _co:i2.Spinner = _v.component;
    const currVal_0:any = _co._c;
    _ck(_v,1,0,currVal_0);
    const currVal_1:any = _co._l;
    _ck(_v,3,0,currVal_1);
  },(null as any));
}
export function View_Spinner_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'ion-spinner',
      ([] as any[]),[[2,'spinner-paused',(null as any)]],(null as any),(null as any),
      View_Spinner_0,RenderType_Spinner)),i0.ɵdid(114688,(null as any),0,i2.Spinner,
      [i3.Config,i0.ElementRef,i0.Renderer],(null as any),(null as any))],(_ck,_v) => {
    _ck(_v,1,0);
  },(_ck,_v) => {
    const currVal_0:any = i0.ɵnov(_v,1)._paused;
    _ck(_v,0,0,currVal_0);
  });
}
export const SpinnerNgFactory:i0.ComponentFactory<i2.Spinner> = i0.ɵccf('ion-spinner',
    i2.Spinner,View_Spinner_Host_0,{color:'color',mode:'mode',name:'name',duration:'duration',
        paused:'paused'},{},([] as any[]));
