// ionic-angular/components/action-sheet/action-sheet-component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from 'ionic-angular/components/icon/icon';
import * as i2 from 'ionic-angular/config/config';
import * as i3 from '../button/button.ngfactory';
import * as i4 from '@angular/common';
import * as i5 from 'ionic-angular/components/button/button';
import * as i6 from 'ionic-angular/components/action-sheet/action-sheet-component';
import * as i7 from 'ionic-angular/components/backdrop/backdrop';
import * as i8 from 'ionic-angular/navigation/view-controller';
import * as i9 from 'ionic-angular/gestures/gesture-controller';
import * as i10 from 'ionic-angular/navigation/nav-params';
const styles_ActionSheetCmp:any[] = ([] as any[]);
export const RenderType_ActionSheetCmp:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_ActionSheetCmp,data:{}});
function View_ActionSheetCmp_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'div',[['class',
      'action-sheet-title']],[[8,'id',0]],(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵted((null as any),['','']))],(null as any),(_ck,
      _v) => {
    var _co:any = _v.component;
    const currVal_0:any = i0.ɵinlineInterpolate(1,'',_co.hdrId,'');
    _ck(_v,0,0,currVal_0);
    const currVal_1:any = _co.d.title;
    _ck(_v,1,0,currVal_1);
  });
}
function View_ActionSheetCmp_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'div',[['class',
      'action-sheet-sub-title']],[[8,'id',0]],(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵted((null as any),['','']))],(null as any),(_ck,
      _v) => {
    var _co:any = _v.component;
    const currVal_0:any = i0.ɵinlineInterpolate(1,'',_co.descId,'');
    _ck(_v,0,0,currVal_0);
    const currVal_1:any = _co.d.subTitle;
    _ck(_v,1,0,currVal_1);
  });
}
function View_ActionSheetCmp_4(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'ion-icon',[['class',
      'action-sheet-icon'],['role','img']],[[2,'hide',(null as any)]],(null as any),
      (null as any),(null as any),(null as any))),i0.ɵdid(147456,(null as any),0,i1.Icon,
      [i2.Config,i0.ElementRef,i0.Renderer],{name:[0,'name']},(null as any))],(_ck,
      _v) => {
    const currVal_1:any = (<any>_v.parent).context.$implicit.icon;
    _ck(_v,1,0,currVal_1);
  },(_ck,_v) => {
    const currVal_0:any = i0.ɵnov(_v,1)._hidden;
    _ck(_v,0,0,currVal_0);
  });
}
function View_ActionSheetCmp_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),5,'button',[['class',
      'disable-hover'],['ion-button','action-sheet-button']],[[1,'icon-start',0]],
      [[(null as any),'click']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_co.click(_v.context.$implicit)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },i3.View_Button_0,i3.RenderType_Button)),i0.ɵdid(278528,(null as any),0,i4.NgClass,
      [i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,
          'klass'],ngClass:[1,'ngClass']},(null as any)),i0.ɵdid(1097728,(null as any),
      0,i5.Button,[[8,'action-sheet-button'],i2.Config,i0.ElementRef,i0.Renderer],
      (null as any),(null as any)),(_l()(),i0.ɵand(16777216,(null as any),0,1,(null as any),
      View_ActionSheetCmp_4)),i0.ɵdid(16384,(null as any),0,i4.NgIf,[i0.ViewContainerRef,
      i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵted(0,['','']))],
      (_ck,_v) => {
        const currVal_1:any = 'disable-hover';
        const currVal_2:any = _v.context.$implicit.cssClass;
        _ck(_v,1,0,currVal_1,currVal_2);
        const currVal_3:any = _v.context.$implicit.icon;
        _ck(_v,4,0,currVal_3);
      },(_ck,_v) => {
        const currVal_0:any = (_v.context.$implicit.icon? '': (null as any));
        _ck(_v,0,0,currVal_0);
        const currVal_4:any = _v.context.$implicit.text;
        _ck(_v,5,0,currVal_4);
      });
}
function View_ActionSheetCmp_6(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'ion-icon',[['class',
      'action-sheet-icon'],['role','img']],[[2,'hide',(null as any)]],(null as any),
      (null as any),(null as any),(null as any))),i0.ɵdid(147456,(null as any),0,i1.Icon,
      [i2.Config,i0.ElementRef,i0.Renderer],{name:[0,'name']},(null as any))],(_ck,
      _v) => {
    var _co:any = _v.component;
    const currVal_1:any = _co.cancelButton.icon;
    _ck(_v,1,0,currVal_1);
  },(_ck,_v) => {
    const currVal_0:any = i0.ɵnov(_v,1)._hidden;
    _ck(_v,0,0,currVal_0);
  });
}
function View_ActionSheetCmp_5(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),6,'div',[['class',
      'action-sheet-group']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵeld(0,(null as any),(null as any),5,'button',[['class',
      'action-sheet-cancel disable-hover'],['ion-button','action-sheet-button']],[[1,
      'icon-start',0]],[[(null as any),'click']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>_co.click(_co.cancelButton)) !== false);
      ad = (pd_0 && ad);
    }
    return ad;
  },i3.View_Button_0,i3.RenderType_Button)),i0.ɵdid(278528,(null as any),0,i4.NgClass,
      [i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,
          'klass'],ngClass:[1,'ngClass']},(null as any)),i0.ɵdid(1097728,(null as any),
      0,i5.Button,[[8,'action-sheet-button'],i2.Config,i0.ElementRef,i0.Renderer],
      (null as any),(null as any)),(_l()(),i0.ɵand(16777216,(null as any),0,1,(null as any),
      View_ActionSheetCmp_6)),i0.ɵdid(16384,(null as any),0,i4.NgIf,[i0.ViewContainerRef,
      i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵted(0,['','']))],
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_1:any = 'action-sheet-cancel disable-hover';
        const currVal_2:any = _co.cancelButton.cssClass;
        _ck(_v,2,0,currVal_1,currVal_2);
        const currVal_3:any = _co.cancelButton.icon;
        _ck(_v,5,0,currVal_3);
      },(_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = (_co.cancelButton.icon? '': (null as any));
        _ck(_v,1,0,currVal_0);
        const currVal_4:any = _co.cancelButton.text;
        _ck(_v,6,0,currVal_4);
      });
}
export function View_ActionSheetCmp_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'ion-backdrop',
      [['disable-activated',''],['role','presentation'],['tappable','']],[[2,'backdrop-no-tappable',
          (null as any)]],[[(null as any),'click']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:i6.ActionSheetCmp = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_co.bdClick()) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),i0.ɵdid(16384,(null as any),0,i7.Backdrop,[i0.ElementRef,
      i0.Renderer],(null as any),(null as any)),(_l()(),i0.ɵeld(0,(null as any),(null as any),
      10,'div',[['class','action-sheet-wrapper']],(null as any),(null as any),(null as any),
      (null as any),(null as any))),(_l()(),i0.ɵeld(0,(null as any),(null as any),
      9,'div',[['class','action-sheet-container']],(null as any),(null as any),(null as any),
      (null as any),(null as any))),(_l()(),i0.ɵeld(0,(null as any),(null as any),
      6,'div',[['class','action-sheet-group']],(null as any),(null as any),(null as any),
      (null as any),(null as any))),(_l()(),i0.ɵand(16777216,(null as any),(null as any),
      1,(null as any),View_ActionSheetCmp_1)),i0.ɵdid(16384,(null as any),0,i4.NgIf,
      [i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),
      i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_ActionSheetCmp_2)),
      i0.ɵdid(16384,(null as any),0,i4.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵand(16777216,(null as any),(null as any),
          1,(null as any),View_ActionSheetCmp_3)),i0.ɵdid(802816,(null as any),0,i4.NgForOf,
          [i0.ViewContainerRef,i0.TemplateRef,i0.IterableDiffers],{ngForOf:[0,'ngForOf']},
          (null as any)),(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),
          View_ActionSheetCmp_5)),i0.ɵdid(16384,(null as any),0,i4.NgIf,[i0.ViewContainerRef,
          i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any))],(_ck,_v) => {
    var _co:i6.ActionSheetCmp = _v.component;
    const currVal_1:any = _co.d.title;
    _ck(_v,6,0,currVal_1);
    const currVal_2:any = _co.d.subTitle;
    _ck(_v,8,0,currVal_2);
    const currVal_3:any = _co.d.buttons;
    _ck(_v,10,0,currVal_3);
    const currVal_4:any = _co.cancelButton;
    _ck(_v,12,0,currVal_4);
  },(_ck,_v) => {
    var _co:i6.ActionSheetCmp = _v.component;
    const currVal_0:boolean = !_co.d.enableBackdropDismiss;
    _ck(_v,0,0,currVal_0);
  });
}
export function View_ActionSheetCmp_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'ion-action-sheet',
      [['role','dialog']],[[1,'aria-labelledby',0],[1,'aria-describedby',0]],[['body',
          'keyup']],(_v,en,$event) => {
        var ad:boolean = true;
        if (('body:keyup' === en)) {
          const pd_0:any = ((<any>i0.ɵnov(_v,1).keyUp($event)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },View_ActionSheetCmp_0,RenderType_ActionSheetCmp)),i0.ɵdid(180224,(null as any),
      0,i6.ActionSheetCmp,[i8.ViewController,i2.Config,i0.ElementRef,i9.GestureController,
          i10.NavParams,i0.Renderer],(null as any),(null as any))],(null as any),(_ck,
      _v) => {
    const currVal_0:any = i0.ɵnov(_v,1).hdrId;
    const currVal_1:any = i0.ɵnov(_v,1).descId;
    _ck(_v,0,0,currVal_0,currVal_1);
  });
}
export const ActionSheetCmpNgFactory:i0.ComponentFactory<i6.ActionSheetCmp> = i0.ɵccf('ion-action-sheet',
    i6.ActionSheetCmp,View_ActionSheetCmp_Host_0,{},{},([] as any[]));
