// ionic-angular/components/alert/alert-component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '../button/button.ngfactory';
import * as i2 from 'ionic-angular/components/button/button';
import * as i3 from 'ionic-angular/config/config';
import * as i4 from '@angular/common';
import * as i5 from '@angular/forms';
import * as i6 from 'ionic-angular/components/alert/alert-component';
import * as i7 from 'ionic-angular/components/backdrop/backdrop';
import * as i8 from 'ionic-angular/navigation/view-controller';
import * as i9 from 'ionic-angular/gestures/gesture-controller';
import * as i10 from 'ionic-angular/navigation/nav-params';
import * as i11 from 'ionic-angular/platform/platform';
const styles_AlertCmp:any[] = ([] as any[]);
export const RenderType_AlertCmp:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_AlertCmp,
    data:{}});
function View_AlertCmp_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),0,'h2',[['class',
      'alert-title']],[[8,'id',0],[8,'innerHTML',1]],(null as any),(null as any),(null as any),
      (null as any)))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = i0.ɵinlineInterpolate(1,'',_co.hdrId,'');
    const currVal_1:any = _co.d.title;
    _ck(_v,0,0,currVal_0,currVal_1);
  });
}
function View_AlertCmp_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),0,'h3',[['class',
      'alert-sub-title']],[[8,'id',0],[8,'innerHTML',1]],(null as any),(null as any),
      (null as any),(null as any)))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = i0.ɵinlineInterpolate(1,'',_co.subHdrId,'');
    const currVal_1:any = _co.d.subTitle;
    _ck(_v,0,0,currVal_0,currVal_1);
  });
}
function View_AlertCmp_5(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),5,'button',[['class',
      'alert-tappable alert-radio'],['ion-button','alert-radio-button'],['role','radio']],
      [[1,'aria-checked',0],[8,'disabled',0],[1,'id',0]],[[(null as any),'click']],
      (_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_co.rbClick(_v.context.$implicit)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },i1.View_Button_0,i1.RenderType_Button)),i0.ɵdid(1097728,(null as any),0,i2.Button,
      [[8,'alert-radio-button'],i3.Config,i0.ElementRef,i0.Renderer],(null as any),
      (null as any)),(_l()(),i0.ɵeld(0,(null as any),0,1,'div',[['class','alert-radio-icon']],
      (null as any),(null as any),(null as any),(null as any),(null as any))),(_l()(),
      i0.ɵeld(0,(null as any),(null as any),0,'div',[['class','alert-radio-inner']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵeld(0,(null as any),0,1,'div',[['class','alert-radio-label']],(null as any),
          (null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['','']))],(null as any),(_ck,_v) => {
    const currVal_0:any = _v.context.$implicit.checked;
    const currVal_1:any = _v.context.$implicit.disabled;
    const currVal_2:any = _v.context.$implicit.id;
    _ck(_v,0,0,currVal_0,currVal_1,currVal_2);
    const currVal_3:any = _v.context.$implicit.label;
    _ck(_v,5,0,currVal_3);
  });
}
function View_AlertCmp_4(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'div',[['class',
      'alert-radio-group'],['role','radiogroup']],[[1,'aria-labelledby',0],[1,'aria-activedescendant',
      0]],(null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵand(16777216,
      (null as any),(null as any),1,(null as any),View_AlertCmp_5)),i0.ɵdid(802816,
      (null as any),0,i4.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,i0.IterableDiffers],
      {ngForOf:[0,'ngForOf']},(null as any))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_2:any = _co.d.inputs;
    _ck(_v,2,0,currVal_2);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.hdrId;
    const currVal_1:any = _co.activeId;
    _ck(_v,0,0,currVal_0,currVal_1);
  });
}
function View_AlertCmp_7(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),5,'button',[['class',
      'alert-tappable alert-checkbox'],['ion-button','alert-checkbox-button'],['role',
      'checkbox']],[[1,'aria-checked',0],[1,'id',0],[8,'disabled',0]],[[(null as any),
      'click']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>_co.cbClick(_v.context.$implicit)) !== false);
      ad = (pd_0 && ad);
    }
    return ad;
  },i1.View_Button_0,i1.RenderType_Button)),i0.ɵdid(1097728,(null as any),0,i2.Button,
      [[8,'alert-checkbox-button'],i3.Config,i0.ElementRef,i0.Renderer],(null as any),
      (null as any)),(_l()(),i0.ɵeld(0,(null as any),0,1,'div',[['class','alert-checkbox-icon']],
      (null as any),(null as any),(null as any),(null as any),(null as any))),(_l()(),
      i0.ɵeld(0,(null as any),(null as any),0,'div',[['class','alert-checkbox-inner']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵeld(0,(null as any),0,1,'div',[['class','alert-checkbox-label']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['','']))],(null as any),(_ck,_v) => {
    const currVal_0:any = _v.context.$implicit.checked;
    const currVal_1:any = _v.context.$implicit.id;
    const currVal_2:any = _v.context.$implicit.disabled;
    _ck(_v,0,0,currVal_0,currVal_1,currVal_2);
    const currVal_3:any = _v.context.$implicit.label;
    _ck(_v,5,0,currVal_3);
  });
}
function View_AlertCmp_6(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'div',[['class',
      'alert-checkbox-group']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),
      View_AlertCmp_7)),i0.ɵdid(802816,(null as any),0,i4.NgForOf,[i0.ViewContainerRef,
      i0.TemplateRef,i0.IterableDiffers],{ngForOf:[0,'ngForOf']},(null as any))],(_ck,
      _v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.d.inputs;
    _ck(_v,2,0,currVal_0);
  },(null as any));
}
function View_AlertCmp_9(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),6,'div',[['class',
      'alert-input-wrapper']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵeld(0,(null as any),(null as any),5,'input',[['class',
      'alert-input']],[[8,'placeholder',0],[8,'type',0],[8,'min',0],[8,'max',0],[1,
      'id',0],[2,'ng-untouched',(null as any)],[2,'ng-touched',(null as any)],[2,'ng-pristine',
      (null as any)],[2,'ng-dirty',(null as any)],[2,'ng-valid',(null as any)],[2,
      'ng-invalid',(null as any)],[2,'ng-pending',(null as any)]],[[(null as any),
      'ngModelChange'],[(null as any),'input'],[(null as any),'blur'],[(null as any),
      'compositionstart'],[(null as any),'compositionend']],(_v,en,$event) => {
    var ad:boolean = true;
    if (('input' === en)) {
      const pd_0:any = ((<any>i0.ɵnov(_v,2)._handleInput($event.target.value)) !== false);
      ad = (pd_0 && ad);
    }
    if (('blur' === en)) {
      const pd_1:any = ((<any>i0.ɵnov(_v,2).onTouched()) !== false);
      ad = (pd_1 && ad);
    }
    if (('compositionstart' === en)) {
      const pd_2:any = ((<any>i0.ɵnov(_v,2)._compositionStart()) !== false);
      ad = (pd_2 && ad);
    }
    if (('compositionend' === en)) {
      const pd_3:any = ((<any>i0.ɵnov(_v,2)._compositionEnd($event.target.value)) !== false);
      ad = (pd_3 && ad);
    }
    if (('ngModelChange' === en)) {
      const pd_4:any = ((<any>(_v.context.$implicit.value = $event)) !== false);
      ad = (pd_4 && ad);
    }
    return ad;
  },(null as any),(null as any))),i0.ɵdid(16384,(null as any),0,i5.DefaultValueAccessor,
      [i0.Renderer2,i0.ElementRef,[2,i5.COMPOSITION_BUFFER_MODE]],(null as any),(null as any)),
      i0.ɵprd(1024,(null as any),i5.NG_VALUE_ACCESSOR,(p0_0:any) => {
        return [p0_0];
      },[i5.DefaultValueAccessor]),i0.ɵdid(671744,(null as any),0,i5.NgModel,[[8,(null as any)],
          [8,(null as any)],[8,(null as any)],[2,i5.NG_VALUE_ACCESSOR]],{model:[0,
          'model']},{update:'ngModelChange'}),i0.ɵprd(2048,(null as any),i5.NgControl,
          (null as any),[i5.NgModel]),i0.ɵdid(16384,(null as any),0,i5.NgControlStatus,
          [i5.NgControl],(null as any),(null as any))],(_ck,_v) => {
    const currVal_12:any = _v.context.$implicit.value;
    _ck(_v,4,0,currVal_12);
  },(_ck,_v) => {
    const currVal_0:any = _v.context.$implicit.placeholder;
    const currVal_1:any = _v.context.$implicit.type;
    const currVal_2:any = _v.context.$implicit.min;
    const currVal_3:any = _v.context.$implicit.max;
    const currVal_4:any = _v.context.$implicit.id;
    const currVal_5:any = i0.ɵnov(_v,6).ngClassUntouched;
    const currVal_6:any = i0.ɵnov(_v,6).ngClassTouched;
    const currVal_7:any = i0.ɵnov(_v,6).ngClassPristine;
    const currVal_8:any = i0.ɵnov(_v,6).ngClassDirty;
    const currVal_9:any = i0.ɵnov(_v,6).ngClassValid;
    const currVal_10:any = i0.ɵnov(_v,6).ngClassInvalid;
    const currVal_11:any = i0.ɵnov(_v,6).ngClassPending;
    _ck(_v,1,1,[currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6,
        currVal_7,currVal_8,currVal_9,currVal_10,currVal_11]);
  });
}
function View_AlertCmp_8(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'div',[['class',
      'alert-input-group']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),
      View_AlertCmp_9)),i0.ɵdid(802816,(null as any),0,i4.NgForOf,[i0.ViewContainerRef,
      i0.TemplateRef,i0.IterableDiffers],{ngForOf:[0,'ngForOf']},(null as any))],(_ck,
      _v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.d.inputs;
    _ck(_v,2,0,currVal_0);
  },(null as any));
}
function View_AlertCmp_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),7,'div',([] as any[]),
      (null as any),(null as any),(null as any),(null as any),(null as any))),i0.ɵdid(16384,
      (null as any),0,i4.NgSwitch,([] as any[]),{ngSwitch:[0,'ngSwitch']},(null as any)),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_AlertCmp_4)),
      i0.ɵdid(278528,(null as any),0,i4.NgSwitchCase,[i0.ViewContainerRef,i0.TemplateRef,
          i4.NgSwitch],{ngSwitchCase:[0,'ngSwitchCase']},(null as any)),(_l()(),i0.ɵand(16777216,
          (null as any),(null as any),1,(null as any),View_AlertCmp_6)),i0.ɵdid(278528,
          (null as any),0,i4.NgSwitchCase,[i0.ViewContainerRef,i0.TemplateRef,i4.NgSwitch],
          {ngSwitchCase:[0,'ngSwitchCase']},(null as any)),(_l()(),i0.ɵand(16777216,
          (null as any),(null as any),1,(null as any),View_AlertCmp_8)),i0.ɵdid(16384,
          (null as any),0,i4.NgSwitchDefault,[i0.ViewContainerRef,i0.TemplateRef,i4.NgSwitch],
          (null as any),(null as any))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.inputType;
    _ck(_v,1,0,currVal_0);
    const currVal_1:any = 'radio';
    _ck(_v,3,0,currVal_1);
    const currVal_2:any = 'checkbox';
    _ck(_v,5,0,currVal_2);
  },(null as any));
}
function View_AlertCmp_10(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),3,'button',[['ion-button',
      'alert-button']],(null as any),[[(null as any),'click']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>_co.btnClick(_v.context.$implicit)) !== false);
      ad = (pd_0 && ad);
    }
    return ad;
  },i1.View_Button_0,i1.RenderType_Button)),i0.ɵdid(278528,(null as any),0,i4.NgClass,
      [i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{ngClass:[0,
          'ngClass']},(null as any)),i0.ɵdid(1097728,(null as any),0,i2.Button,[[8,
      'alert-button'],i3.Config,i0.ElementRef,i0.Renderer],(null as any),(null as any)),
      (_l()(),i0.ɵted(0,['','']))],(_ck,_v) => {
    const currVal_0:any = _v.context.$implicit.cssClass;
    _ck(_v,1,0,currVal_0);
  },(_ck,_v) => {
    const currVal_1:any = _v.context.$implicit.text;
    _ck(_v,3,0,currVal_1);
  });
}
export function View_AlertCmp_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'ion-backdrop',
      [['disable-activated',''],['role','presentation'],['tappable','']],[[2,'backdrop-no-tappable',
          (null as any)]],[[(null as any),'click']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:i6.AlertCmp = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_co.bdClick()) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),i0.ɵdid(16384,(null as any),0,i7.Backdrop,[i0.ElementRef,
      i0.Renderer],(null as any),(null as any)),(_l()(),i0.ɵeld(0,(null as any),(null as any),
      13,'div',[['class','alert-wrapper']],(null as any),(null as any),(null as any),
      (null as any),(null as any))),(_l()(),i0.ɵeld(0,(null as any),(null as any),
      4,'div',[['class','alert-head']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),
      View_AlertCmp_1)),i0.ɵdid(16384,(null as any),0,i4.NgIf,[i0.ViewContainerRef,
      i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵand(16777216,(null as any),
      (null as any),1,(null as any),View_AlertCmp_2)),i0.ɵdid(16384,(null as any),
      0,i4.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),0,'div',[['class','alert-message']],
          [[8,'id',0],[8,'innerHTML',1]],(null as any),(null as any),(null as any),
          (null as any))),(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),
          View_AlertCmp_3)),i0.ɵdid(16384,(null as any),0,i4.NgIf,[i0.ViewContainerRef,
          i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),4,'div',[['class','alert-button-group']],(null as any),(null as any),
          (null as any),(null as any),(null as any))),i0.ɵdid(278528,(null as any),
          0,i4.NgClass,[i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],
          {klass:[0,'klass'],ngClass:[1,'ngClass']},(null as any)),i0.ɵpod({'alert-button-group-vertical':0}),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_AlertCmp_10)),
      i0.ɵdid(802816,(null as any),0,i4.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,
          i0.IterableDiffers],{ngForOf:[0,'ngForOf']},(null as any))],(_ck,_v) => {
    var _co:i6.AlertCmp = _v.component;
    const currVal_1:any = _co.d.title;
    _ck(_v,5,0,currVal_1);
    const currVal_2:any = _co.d.subTitle;
    _ck(_v,7,0,currVal_2);
    const currVal_5:any = _co.d.inputs.length;
    _ck(_v,10,0,currVal_5);
    const currVal_6:any = 'alert-button-group';
    const currVal_7:any = _ck(_v,13,0,(_co.d.buttons.length > 2));
    _ck(_v,12,0,currVal_6,currVal_7);
    const currVal_8:any = _co.d.buttons;
    _ck(_v,15,0,currVal_8);
  },(_ck,_v) => {
    var _co:i6.AlertCmp = _v.component;
    const currVal_0:boolean = !_co.d.enableBackdropDismiss;
    _ck(_v,0,0,currVal_0);
    const currVal_3:any = i0.ɵinlineInterpolate(1,'',_co.msgId,'');
    const currVal_4:any = _co.d.message;
    _ck(_v,8,0,currVal_3,currVal_4);
  });
}
export function View_AlertCmp_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'ion-alert',[['role',
      'dialog']],[[1,'aria-labelledby',0],[1,'aria-describedby',0]],[['body','keyup']],
      (_v,en,$event) => {
        var ad:boolean = true;
        if (('body:keyup' === en)) {
          const pd_0:any = ((<any>i0.ɵnov(_v,1).keyUp($event)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },View_AlertCmp_0,RenderType_AlertCmp)),i0.ɵdid(180224,(null as any),0,i6.AlertCmp,
      [i8.ViewController,i0.ElementRef,i3.Config,i9.GestureController,i10.NavParams,
          i0.Renderer,i11.Platform],(null as any),(null as any))],(null as any),(_ck,
      _v) => {
    const currVal_0:any = i0.ɵnov(_v,1).hdrId;
    const currVal_1:any = i0.ɵnov(_v,1).descId;
    _ck(_v,0,0,currVal_0,currVal_1);
  });
}
export const AlertCmpNgFactory:i0.ComponentFactory<i6.AlertCmp> = i0.ɵccf('ion-alert',
    i6.AlertCmp,View_AlertCmp_Host_0,{},{},([] as any[]));
