// ionic-angular/components/loading/loading-component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '../spinner/spinner.ngfactory';
import * as i2 from 'ionic-angular/components/spinner/spinner';
import * as i3 from 'ionic-angular/config/config';
import * as i4 from 'ionic-angular/components/loading/loading-component';
import * as i5 from 'ionic-angular/components/backdrop/backdrop';
import * as i6 from '@angular/common';
import * as i7 from 'ionic-angular/navigation/view-controller';
import * as i8 from 'ionic-angular/gestures/gesture-controller';
import * as i9 from 'ionic-angular/navigation/nav-params';
const styles_LoadingCmp:any[] = ([] as any[]);
export const RenderType_LoadingCmp:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_LoadingCmp,
    data:{}});
function View_LoadingCmp_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'div',[['class',
      'loading-spinner']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'ion-spinner',
      ([] as any[]),[[2,'spinner-paused',(null as any)]],(null as any),(null as any),
      i1.View_Spinner_0,i1.RenderType_Spinner)),i0.ɵdid(114688,(null as any),0,i2.Spinner,
      [i3.Config,i0.ElementRef,i0.Renderer],{name:[0,'name']},(null as any))],(_ck,
      _v) => {
    var _co:any = _v.component;
    const currVal_1:any = _co.d.spinner;
    _ck(_v,2,0,currVal_1);
  },(_ck,_v) => {
    const currVal_0:any = i0.ɵnov(_v,2)._paused;
    _ck(_v,1,0,currVal_0);
  });
}
function View_LoadingCmp_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),0,'div',[['class',
      'loading-content']],[[8,'innerHTML',1]],(null as any),(null as any),(null as any),
      (null as any)))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.d.content;
    _ck(_v,0,0,currVal_0);
  });
}
export function View_LoadingCmp_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'ion-backdrop',
      [['disable-activated',''],['role','presentation'],['tappable','']],[[8,'hidden',
          0],[2,'backdrop-no-tappable',(null as any)]],[[(null as any),'click']],(_v,
          en,$event) => {
        var ad:boolean = true;
        var _co:i4.LoadingCmp = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_co.bdClick()) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),i0.ɵdid(16384,(null as any),0,i5.Backdrop,[i0.ElementRef,
      i0.Renderer],(null as any),(null as any)),(_l()(),i0.ɵeld(0,(null as any),(null as any),
      4,'div',[['class','loading-wrapper']],(null as any),(null as any),(null as any),
      (null as any),(null as any))),(_l()(),i0.ɵand(16777216,(null as any),(null as any),
      1,(null as any),View_LoadingCmp_1)),i0.ɵdid(16384,(null as any),0,i6.NgIf,[i0.ViewContainerRef,
      i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵand(16777216,(null as any),
      (null as any),1,(null as any),View_LoadingCmp_2)),i0.ɵdid(16384,(null as any),
      0,i6.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any))],
      (_ck,_v) => {
        var _co:i4.LoadingCmp = _v.component;
        const currVal_2:any = _co.showSpinner;
        _ck(_v,4,0,currVal_2);
        const currVal_3:any = _co.d.content;
        _ck(_v,6,0,currVal_3);
      },(_ck,_v) => {
        var _co:i4.LoadingCmp = _v.component;
        const currVal_0:boolean = !_co.d.showBackdrop;
        const currVal_1:boolean = !_co.d.enableBackdropDismiss;
        _ck(_v,0,0,currVal_0,currVal_1);
      });
}
export function View_LoadingCmp_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'ion-loading',
      [['role','dialog']],(null as any),[['body','keyup']],(_v,en,$event) => {
        var ad:boolean = true;
        if (('body:keyup' === en)) {
          const pd_0:any = ((<any>i0.ɵnov(_v,1).keyUp($event)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },View_LoadingCmp_0,RenderType_LoadingCmp)),i0.ɵdid(245760,(null as any),0,i4.LoadingCmp,
      [i7.ViewController,i3.Config,i0.ElementRef,i8.GestureController,i9.NavParams,
          i0.Renderer],(null as any),(null as any))],(_ck,_v) => {
    _ck(_v,1,0);
  },(null as any));
}
export const LoadingCmpNgFactory:i0.ComponentFactory<i4.LoadingCmp> = i0.ɵccf('ion-loading',
    i4.LoadingCmp,View_LoadingCmp_Host_0,{},{},([] as any[]));
