// ionic-angular/components/input/input.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/common';
import * as i2 from '../button/button.ngfactory';
import * as i3 from 'ionic-angular/components/button/button';
import * as i4 from 'ionic-angular/config/config';
import * as i5 from 'ionic-angular/components/input/input';
import * as i6 from 'ionic-angular/platform/platform';
import * as i7 from 'ionic-angular/util/form';
import * as i8 from 'ionic-angular/components/app/app';
import * as i9 from 'ionic-angular/components/content/content';
import * as i10 from 'ionic-angular/components/item/item';
import * as i11 from '@angular/forms';
import * as i12 from 'ionic-angular/platform/dom-controller';
const styles_TextInput:any[] = ([] as any[]);
export const RenderType_TextInput:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_TextInput,
    data:{}});
function View_TextInput_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,[[1,0],['textInput',1]],(null as any),1,'input',
      [['class','text-input']],[[8,'type',0],[1,'aria-labelledby',0],[1,'min',0],[1,
          'max',0],[1,'step',0],[1,'autocomplete',0],[1,'autocorrect',0],[8,'placeholder',
          0],[8,'disabled',0],[8,'readOnly',0]],[[(null as any),'input'],[(null as any),
          'blur'],[(null as any),'focus'],[(null as any),'keydown']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('input' === en)) {
          const pd_0:any = ((<any>_co.onInput($event)) !== false);
          ad = (pd_0 && ad);
        }
        if (('blur' === en)) {
          const pd_1:any = ((<any>_co.onBlur($event)) !== false);
          ad = (pd_1 && ad);
        }
        if (('focus' === en)) {
          const pd_2:any = ((<any>_co.onFocus($event)) !== false);
          ad = (pd_2 && ad);
        }
        if (('keydown' === en)) {
          const pd_3:any = ((<any>_co.onKeydown($event)) !== false);
          ad = (pd_3 && ad);
        }
        return ad;
      },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i1.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
      'ngClass']},(null as any))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_10:any = 'text-input';
    const currVal_11:any = ('text-input-' + _co._mode);
    _ck(_v,1,0,currVal_10,currVal_11);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co._type;
    const currVal_1:any = _co._labelId;
    const currVal_2:any = _co.min;
    const currVal_3:any = _co.max;
    const currVal_4:any = _co.step;
    const currVal_5:any = _co.autocomplete;
    const currVal_6:any = _co.autocorrect;
    const currVal_7:any = _co.placeholder;
    const currVal_8:any = _co._disabled;
    const currVal_9:any = _co._readonly;
    _ck(_v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6,
        currVal_7,currVal_8,currVal_9);
  });
}
function View_TextInput_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,[[1,0],['textInput',1]],(null as any),1,'textarea',
      [['class','text-input']],[[1,'aria-labelledby',0],[1,'autocomplete',0],[1,'autocorrect',
          0],[8,'placeholder',0],[8,'disabled',0],[8,'readOnly',0]],[[(null as any),
          'input'],[(null as any),'blur'],[(null as any),'focus'],[(null as any),'keydown']],
      (_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('input' === en)) {
          const pd_0:any = ((<any>_co.onInput($event)) !== false);
          ad = (pd_0 && ad);
        }
        if (('blur' === en)) {
          const pd_1:any = ((<any>_co.onBlur($event)) !== false);
          ad = (pd_1 && ad);
        }
        if (('focus' === en)) {
          const pd_2:any = ((<any>_co.onFocus($event)) !== false);
          ad = (pd_2 && ad);
        }
        if (('keydown' === en)) {
          const pd_3:any = ((<any>_co.onKeydown($event)) !== false);
          ad = (pd_3 && ad);
        }
        return ad;
      },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i1.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
      'ngClass']},(null as any))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_6:any = 'text-input';
    const currVal_7:any = ('text-input-' + _co._mode);
    _ck(_v,1,0,currVal_6,currVal_7);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co._labelId;
    const currVal_1:any = _co.autocomplete;
    const currVal_2:any = _co.autocorrect;
    const currVal_3:any = _co.placeholder;
    const currVal_4:any = _co._disabled;
    const currVal_5:any = _co._readonly;
    _ck(_v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5);
  });
}
function View_TextInput_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'button',[['class',
      'text-input-clear-icon'],['clear',''],['ion-button',''],['tabindex','-1'],['type',
      'button']],(null as any),[[(null as any),'click'],[(null as any),'mousedown']],
      (_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_co.clearTextInput($event)) !== false);
          ad = (pd_0 && ad);
        }
        if (('mousedown' === en)) {
          const pd_1:any = ((<any>_co.clearTextInput($event)) !== false);
          ad = (pd_1 && ad);
        }
        return ad;
      },i2.View_Button_0,i2.RenderType_Button)),i0.ɵdid(1097728,(null as any),0,i3.Button,
      [[8,''],i4.Config,i0.ElementRef,i0.Renderer],{clear:[0,'clear']},(null as any))],
      (_ck,_v) => {
        const currVal_0:any = '';
        _ck(_v,1,0,currVal_0);
      },(null as any));
}
function View_TextInput_4(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),0,'div',[['class',
      'input-cover']],(null as any),[[(null as any),'touchstart'],[(null as any),'touchend'],
      [(null as any),'mousedown'],[(null as any),'mouseup']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('touchstart' === en)) {
      const pd_0:any = ((<any>_co._pointerStart($event)) !== false);
      ad = (pd_0 && ad);
    }
    if (('touchend' === en)) {
      const pd_1:any = ((<any>_co._pointerEnd($event)) !== false);
      ad = (pd_1 && ad);
    }
    if (('mousedown' === en)) {
      const pd_2:any = ((<any>_co._pointerStart($event)) !== false);
      ad = (pd_2 && ad);
    }
    if (('mouseup' === en)) {
      const pd_3:any = ((<any>_co._pointerEnd($event)) !== false);
      ad = (pd_3 && ad);
    }
    return ad;
  },(null as any),(null as any)))],(null as any),(null as any));
}
export function View_TextInput_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[i0.ɵqud(671088640,1,{_native:0}),(_l()(),i0.ɵand(16777216,(null as any),
      (null as any),1,(null as any),View_TextInput_1)),i0.ɵdid(16384,(null as any),
      0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_TextInput_2)),
      i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵand(16777216,(null as any),(null as any),
          1,(null as any),View_TextInput_3)),i0.ɵdid(16384,(null as any),0,i1.NgIf,
          [i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),
          i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_TextInput_4)),
      i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any))],(_ck,_v) => {
    var _co:i5.TextInput = _v.component;
    const currVal_0:boolean = !_co._isTextarea;
    _ck(_v,2,0,currVal_0);
    const currVal_1:any = _co._isTextarea;
    _ck(_v,4,0,currVal_1);
    const currVal_2:any = _co._clearInput;
    _ck(_v,6,0,currVal_2);
    const currVal_3:any = _co._useAssist;
    _ck(_v,8,0,currVal_3);
  },(null as any));
}
export function View_TextInput_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'ion-input',([] as any[]),
      (null as any),(null as any),(null as any),View_TextInput_0,RenderType_TextInput)),
      i0.ɵdid(5423104,(null as any),0,i5.TextInput,[i4.Config,i6.Platform,i7.Form,
          i8.App,i0.ElementRef,i0.Renderer,[2,i9.Content],[2,i10.Item],[2,i11.NgControl],
          i12.DomController],(null as any),(null as any))],(null as any),(null as any));
}
export const TextInputNgFactory:i0.ComponentFactory<i5.TextInput> = i0.ɵccf('ion-input,ion-textarea',
    i5.TextInput,View_TextInput_Host_0,{value:'value',color:'color',mode:'mode',disabled:'disabled',
        clearInput:'clearInput',type:'type',readonly:'readonly',clearOnEdit:'clearOnEdit',
        autocomplete:'autocomplete',autocorrect:'autocorrect',placeholder:'placeholder',
        min:'min',max:'max',step:'step'},{ionFocus:'ionFocus',ionChange:'ionChange',
        ionBlur:'ionBlur',input:'input',blur:'blur',focus:'focus'},([] as any[]));
