// ionic-angular/components/tabs/tab-button.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from 'ionic-angular/components/icon/icon';
import * as i2 from 'ionic-angular/config/config';
import * as i3 from 'ionic-angular/components/badge/badge';
import * as i4 from '@angular/common';
import * as i5 from 'ionic-angular/components/tabs/tab-button';
const styles_TabButton:any[] = ([] as any[]);
export const RenderType_TabButton:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_TabButton,
    data:{}});
function View_TabButton_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'ion-icon',[['class',
      'tab-button-icon'],['role','img']],[[2,'hide',(null as any)]],(null as any),
      (null as any),(null as any),(null as any))),i0.ɵdid(147456,(null as any),0,i1.Icon,
      [i2.Config,i0.ElementRef,i0.Renderer],{name:[0,'name'],isActive:[1,'isActive']},
      (null as any))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_1:any = _co.tab.tabIcon;
    const currVal_2:any = _co.tab.isSelected;
    _ck(_v,1,0,currVal_1,currVal_2);
  },(_ck,_v) => {
    const currVal_0:any = i0.ɵnov(_v,1)._hidden;
    _ck(_v,0,0,currVal_0);
  });
}
function View_TabButton_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'span',[['class',
      'tab-button-text']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵted((null as any),['','']))],(null as any),(_ck,
      _v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.tab.tabTitle;
    _ck(_v,1,0,currVal_0);
  });
}
function View_TabButton_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'ion-badge',[['class',
      'tab-badge']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      i0.ɵdid(16384,(null as any),0,i3.Badge,[i2.Config,i0.ElementRef,i0.Renderer],
          {color:[0,'color']},(null as any)),(_l()(),i0.ɵted((null as any),['','']))],
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = _co.tab.tabBadgeStyle;
        _ck(_v,1,0,currVal_0);
      },(_ck,_v) => {
        var _co:any = _v.component;
        const currVal_1:any = _co.tab.tabBadge;
        _ck(_v,2,0,currVal_1);
      });
}
export function View_TabButton_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),
      View_TabButton_1)),i0.ɵdid(16384,(null as any),0,i4.NgIf,[i0.ViewContainerRef,
      i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵand(16777216,(null as any),
      (null as any),1,(null as any),View_TabButton_2)),i0.ɵdid(16384,(null as any),
      0,i4.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_TabButton_3)),
      i0.ɵdid(16384,(null as any),0,i4.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵeld(0,(null as any),(null as any),0,
          'div',[['class','button-effect']],(null as any),(null as any),(null as any),
          (null as any),(null as any)))],(_ck,_v) => {
    var _co:i5.TabButton = _v.component;
    const currVal_0:any = _co.tab.tabIcon;
    _ck(_v,1,0,currVal_0);
    const currVal_1:any = _co.tab.tabTitle;
    _ck(_v,3,0,currVal_1);
    const currVal_2:any = _co.tab.tabBadge;
    _ck(_v,5,0,currVal_2);
  },(null as any));
}
export function View_TabButton_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'div',[['class',
      'tab-button']],[[1,'id',0],[1,'aria-controls',0],[1,'aria-selected',0],[2,'has-title',
      (null as any)],[2,'has-icon',(null as any)],[2,'has-title-only',(null as any)],
      [2,'icon-only',(null as any)],[2,'has-badge',(null as any)],[2,'disable-hover',
          (null as any)],[2,'tab-disabled',(null as any)],[2,'tab-hidden',(null as any)]],
      [[(null as any),'click']],(_v,en,$event) => {
        var ad:boolean = true;
        if (('click' === en)) {
          const pd_0:any = ((<any>i0.ɵnov(_v,1).onClick()) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },View_TabButton_0,RenderType_TabButton)),i0.ɵdid(114688,(null as any),0,i5.TabButton,
      [i2.Config,i0.ElementRef,i0.Renderer],(null as any),(null as any))],(_ck,_v) => {
    _ck(_v,1,0);
  },(_ck,_v) => {
    const currVal_0:any = i0.ɵnov(_v,1).tab._btnId;
    const currVal_1:any = i0.ɵnov(_v,1).tab._tabId;
    const currVal_2:any = i0.ɵnov(_v,1).tab.isSelected;
    const currVal_3:any = i0.ɵnov(_v,1).hasTitle;
    const currVal_4:any = i0.ɵnov(_v,1).hasIcon;
    const currVal_5:any = i0.ɵnov(_v,1).hasTitleOnly;
    const currVal_6:any = i0.ɵnov(_v,1).hasIconOnly;
    const currVal_7:any = i0.ɵnov(_v,1).hasBadge;
    const currVal_8:any = i0.ɵnov(_v,1).disHover;
    const currVal_9:boolean = !i0.ɵnov(_v,1).tab.enabled;
    const currVal_10:boolean = !i0.ɵnov(_v,1).tab.show;
    _ck(_v,0,1,[currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6,
        currVal_7,currVal_8,currVal_9,currVal_10]);
  });
}
export const TabButtonNgFactory:i0.ComponentFactory<i5.TabButton> = i0.ɵccf('.tab-button',
    i5.TabButton,View_TabButton_Host_0,{color:'color',mode:'mode',tab:'tab'},{ionSelect:'ionSelect'},
    ([] as any[]));
