// ionic-angular/components/toast/toast-component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '../button/button.ngfactory';
import * as i2 from 'ionic-angular/components/button/button';
import * as i3 from 'ionic-angular/config/config';
import * as i4 from '@angular/common';
import * as i5 from 'ionic-angular/components/toast/toast-component';
import * as i6 from 'ionic-angular/navigation/view-controller';
import * as i7 from 'ionic-angular/navigation/nav-params';
const styles_ToastCmp:any[] = ([] as any[]);
export const RenderType_ToastCmp:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_ToastCmp,
    data:{}});
function View_ToastCmp_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'div',[['class',
      'toast-message']],[[8,'id',0]],(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['','']))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = i0.ɵinlineInterpolate(1,'',_co.hdrId,'');
    _ck(_v,0,0,currVal_0);
    const currVal_1:any = _co.d.message;
    _ck(_v,1,0,currVal_1);
  });
}
function View_ToastCmp_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'button',[['class',
      'toast-button'],['clear',''],['ion-button','']],(null as any),[[(null as any),
      'click']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>_co.cbClick()) !== false);
      ad = (pd_0 && ad);
    }
    return ad;
  },i1.View_Button_0,i1.RenderType_Button)),i0.ɵdid(1097728,(null as any),0,i2.Button,
      [[8,''],i3.Config,i0.ElementRef,i0.Renderer],{clear:[0,'clear']},(null as any)),
      (_l()(),i0.ɵted(0,[' ',' ']))],(_ck,_v) => {
    const currVal_0:any = '';
    _ck(_v,1,0,currVal_0);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_1:any = (_co.d.closeButtonText || 'Close');
    _ck(_v,2,0,currVal_1);
  });
}
export function View_ToastCmp_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),10,'div',[['class',
      'toast-wrapper']],[[2,'toast-bottom',(null as any)],[2,'toast-middle',(null as any)],
      [2,'toast-top',(null as any)]],(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),[' '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),
          7,'div',[['class','toast-container']],(null as any),(null as any),(null as any),
          (null as any),(null as any))),(_l()(),i0.ɵted((null as any),[' '])),(_l()(),
          i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_ToastCmp_1)),
      i0.ɵdid(16384,(null as any),0,i4.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),[' '])),(_l()(),i0.ɵand(16777216,
          (null as any),(null as any),1,(null as any),View_ToastCmp_2)),i0.ɵdid(16384,
          (null as any),0,i4.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},
          (null as any)),(_l()(),i0.ɵted((null as any),[' '])),(_l()(),i0.ɵted((null as any),
          [' ']))],(_ck,_v) => {
    var _co:i5.ToastCmp = _v.component;
    const currVal_3:any = _co.d.message;
    _ck(_v,5,0,currVal_3);
    const currVal_4:any = _co.d.showCloseButton;
    _ck(_v,8,0,currVal_4);
  },(_ck,_v) => {
    var _co:i5.ToastCmp = _v.component;
    const currVal_0:any = (_co.d.position === 'bottom');
    const currVal_1:any = (_co.d.position === 'middle');
    const currVal_2:any = (_co.d.position === 'top');
    _ck(_v,0,0,currVal_0,currVal_1,currVal_2);
  });
}
export function View_ToastCmp_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'ion-toast',[['role',
      'dialog']],[[1,'aria-labelledby',0],[1,'aria-describedby',0]],(null as any),
      (null as any),View_ToastCmp_0,RenderType_ToastCmp)),i0.ɵdid(4243456,(null as any),
      0,i5.ToastCmp,[i6.ViewController,i3.Config,i0.ElementRef,i7.NavParams,i0.Renderer],
      (null as any),(null as any))],(null as any),(_ck,_v) => {
    const currVal_0:any = i0.ɵnov(_v,1).hdrId;
    const currVal_1:any = i0.ɵnov(_v,1).descId;
    _ck(_v,0,0,currVal_0,currVal_1);
  });
}
export const ToastCmpNgFactory:i0.ComponentFactory<i5.ToastCmp> = i0.ɵccf('ion-toast',
    i5.ToastCmp,View_ToastCmp_Host_0,{},{},([] as any[]));
