// ionic-angular/components/datetime/datetime.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/common';
import * as i2 from '../button/button.ngfactory';
import * as i3 from 'ionic-angular/components/button/button';
import * as i4 from 'ionic-angular/config/config';
import * as i5 from 'ionic-angular/components/datetime/datetime';
import * as i6 from '@angular/forms';
import * as i7 from 'ionic-angular/util/form';
import * as i8 from 'ionic-angular/components/item/item';
import * as i9 from 'ionic-angular/components/picker/picker-controller';
const styles_DateTime:any[] = ([] as any[]);
export const RenderType_DateTime:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_DateTime,
    data:{}});
function View_DateTime_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'div',[['class',
      'datetime-text datetime-placeholder']],(null as any),(null as any),(null as any),
      (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['','']))],(null as any),
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = _co.placeholder;
        _ck(_v,1,0,currVal_0);
      });
}
function View_DateTime_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'div',[['class',
      'datetime-text']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['','']))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co._text;
    _ck(_v,1,0,currVal_0);
  });
}
export function View_DateTime_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),
      View_DateTime_1)),i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,
      i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵand(16777216,(null as any),
      (null as any),1,(null as any),View_DateTime_2)),i0.ɵdid(16384,(null as any),
      0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),1,'button',[['aria-haspopup','true'],
          ['class','item-cover'],['ion-button','item-cover'],['type','button']],[[8,
          'id',0],[1,'aria-labelledby',0],[1,'aria-disabled',0]],(null as any),(null as any),
          i2.View_Button_0,i2.RenderType_Button)),i0.ɵdid(1097728,(null as any),0,
          i3.Button,[[8,'item-cover'],i4.Config,i0.ElementRef,i0.Renderer],(null as any),
          (null as any))],(_ck,_v) => {
    var _co:i5.DateTime = _v.component;
    const currVal_0:boolean = !_co._text;
    _ck(_v,1,0,currVal_0);
    const currVal_1:any = _co._text;
    _ck(_v,3,0,currVal_1);
  },(_ck,_v) => {
    var _co:i5.DateTime = _v.component;
    const currVal_2:any = _co.id;
    const currVal_3:any = _co._labelId;
    const currVal_4:any = _co._disabled;
    _ck(_v,4,0,currVal_2,currVal_3,currVal_4);
  });
}
export function View_DateTime_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'ion-datetime',
      ([] as any[]),[[2,'datetime-disabled',(null as any)]],[[(null as any),'click'],
          [(null as any),'keyup.space']],(_v,en,$event) => {
        var ad:boolean = true;
        if (('click' === en)) {
          const pd_0:any = ((<any>i0.ɵnov(_v,2)._click($event)) !== false);
          ad = (pd_0 && ad);
        }
        if (('keyup.space' === en)) {
          const pd_1:any = ((<any>i0.ɵnov(_v,2)._keyup()) !== false);
          ad = (pd_1 && ad);
        }
        return ad;
      },View_DateTime_0,RenderType_DateTime)),i0.ɵprd(5120,(null as any),i6.NG_VALUE_ACCESSOR,
      (p0_0:any) => {
        return [p0_0];
      },[i5.DateTime]),i0.ɵdid(1228800,(null as any),0,i5.DateTime,[i7.Form,i4.Config,
      i0.ElementRef,i0.Renderer,[2,i8.Item],[2,i9.PickerController]],(null as any),
      (null as any))],(null as any),(_ck,_v) => {
    const currVal_0:any = i0.ɵnov(_v,2)._disabled;
    _ck(_v,0,0,currVal_0);
  });
}
export const DateTimeNgFactory:i0.ComponentFactory<i5.DateTime> = i0.ɵccf('ion-datetime',
    i5.DateTime,View_DateTime_Host_0,{color:'color',mode:'mode',disabled:'disabled',
        min:'min',max:'max',displayFormat:'displayFormat',initialValue:'initialValue',
        pickerFormat:'pickerFormat',cancelText:'cancelText',doneText:'doneText',yearValues:'yearValues',
        monthValues:'monthValues',dayValues:'dayValues',hourValues:'hourValues',minuteValues:'minuteValues',
        monthNames:'monthNames',monthShortNames:'monthShortNames',dayNames:'dayNames',
        dayShortNames:'dayShortNames',pickerOptions:'pickerOptions',placeholder:'placeholder'},
    {ionFocus:'ionFocus',ionChange:'ionChange',ionBlur:'ionBlur',ionCancel:'ionCancel'},
    ([] as any[]));
