// ionic-angular/components/range/range.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from './range-knob.ngfactory';
import * as i2 from 'ionic-angular/components/range/range-knob';
import * as i3 from '@angular/common';
import * as i4 from 'ionic-angular/components/range/range';
import * as i5 from '@angular/forms';
import * as i6 from 'ionic-angular/util/form';
import * as i7 from 'ionic-angular/tap-click/haptic';
import * as i8 from 'ionic-angular/components/item/item';
import * as i9 from 'ionic-angular/config/config';
import * as i10 from 'ionic-angular/platform/platform';
import * as i11 from 'ionic-angular/platform/dom-controller';
const styles_Range:any[] = ([] as any[]);
export const RenderType_Range:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_Range,
    data:{}});
function View_Range_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),0,'div',[['class',
      'range-tick'],['role','presentation']],[[4,'left',(null as any)],[2,'range-tick-active',
      (null as any)]],(null as any),(null as any),(null as any),(null as any)))],(null as any),
      (_ck,_v) => {
        const currVal_0:any = _v.context.$implicit.left;
        const currVal_1:any = _v.context.$implicit.active;
        _ck(_v,0,0,currVal_0,currVal_1);
      });
}
function View_Range_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'div',[['class',
      'range-knob-handle'],['role','slider']],[[2,'range-knob-pressed',(null as any)],
      [2,'range-knob-min',(null as any)],[2,'range-knob-max',(null as any)],[4,'left',
          (null as any)],[1,'aria-valuenow',0],[1,'aria-valuemin',0],[1,'aria-valuemax',
          0],[1,'aria-disabled',0],[1,'aria-labelledby',0],[8,'tabIndex',0]],[[(null as any),
      'ionIncrease'],[(null as any),'ionDecrease'],[(null as any),'keydown']],(_v,
      en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('keydown' === en)) {
      const pd_0:any = ((<any>i0.ɵnov(_v,1)._keyup($event)) !== false);
      ad = (pd_0 && ad);
    }
    if (('ionIncrease' === en)) {
      const pd_1:any = ((<any>_co._keyChg(true,true)) !== false);
      ad = (pd_1 && ad);
    }
    if (('ionDecrease' === en)) {
      const pd_2:any = ((<any>_co._keyChg(false,true)) !== false);
      ad = (pd_2 && ad);
    }
    return ad;
  },i1.View_RangeKnob_0,i1.RenderType_RangeKnob)),i0.ɵdid(49152,(null as any),0,i2.RangeKnob,
      ([] as any[]),{ratio:[0,'ratio'],pressed:[1,'pressed'],pin:[2,'pin'],min:[3,
          'min'],max:[4,'max'],val:[5,'val'],disabled:[6,'disabled'],labelId:[7,'labelId']},
      {ionIncrease:'ionIncrease',ionDecrease:'ionDecrease'})],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_10:any = _co._ratioB;
    const currVal_11:any = _co._pressedB;
    const currVal_12:any = _co._pin;
    const currVal_13:any = _co._min;
    const currVal_14:any = _co._max;
    const currVal_15:any = _co._valB;
    const currVal_16:any = _co._disabled;
    const currVal_17:any = _co._labelId;
    _ck(_v,1,0,currVal_10,currVal_11,currVal_12,currVal_13,currVal_14,currVal_15,currVal_16,
        currVal_17);
  },(_ck,_v) => {
    const currVal_0:any = i0.ɵnov(_v,1).pressed;
    const currVal_1:any = ((i0.ɵnov(_v,1).val === i0.ɵnov(_v,1).min) || (i0.ɵnov(_v,
        1).val === (undefined as any)));
    const currVal_2:any = (i0.ɵnov(_v,1).val === i0.ɵnov(_v,1).max);
    const currVal_3:any = i0.ɵnov(_v,1)._x;
    const currVal_4:any = i0.ɵnov(_v,1).val;
    const currVal_5:any = i0.ɵnov(_v,1).min;
    const currVal_6:any = i0.ɵnov(_v,1).max;
    const currVal_7:any = i0.ɵnov(_v,1).disabled;
    const currVal_8:any = i0.ɵnov(_v,1).labelId;
    const currVal_9:any = (i0.ɵnov(_v,1).disabled? (0 - 1): 0);
    _ck(_v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6,
        currVal_7,currVal_8,currVal_9);
  });
}
export function View_Range_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[i0.ɵqud(402653184,1,{_slider:0}),i0.ɵncd((null as any),0),(_l()(),
      i0.ɵeld(0,[[1,0],['slider',1]],(null as any),8,'div',[['class','range-slider']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_Range_1)),
      i0.ɵdid(802816,(null as any),0,i3.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,
          i0.IterableDiffers],{ngForOf:[0,'ngForOf']},(null as any)),(_l()(),i0.ɵeld(0,
          (null as any),(null as any),0,'div',[['class','range-bar'],['role','presentation']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵeld(0,[['bar',1]],(null as any),0,'div',[['class','range-bar range-bar-active'],
          ['role','presentation']],[[4,'left',(null as any)],[4,'right',(null as any)]],
          (null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵeld(0,
          (null as any),(null as any),1,'div',[['class','range-knob-handle'],['role',
              'slider']],[[2,'range-knob-pressed',(null as any)],[2,'range-knob-min',
              (null as any)],[2,'range-knob-max',(null as any)],[4,'left',(null as any)],
              [1,'aria-valuenow',0],[1,'aria-valuemin',0],[1,'aria-valuemax',0],[1,
                  'aria-disabled',0],[1,'aria-labelledby',0],[8,'tabIndex',0]],[[(null as any),
              'ionIncrease'],[(null as any),'ionDecrease'],[(null as any),'keydown']],
          (_v,en,$event) => {
            var ad:boolean = true;
            var _co:i4.Range = _v.component;
            if (('keydown' === en)) {
              const pd_0:any = ((<any>i0.ɵnov(_v,8)._keyup($event)) !== false);
              ad = (pd_0 && ad);
            }
            if (('ionIncrease' === en)) {
              const pd_1:any = ((<any>_co._keyChg(true,false)) !== false);
              ad = (pd_1 && ad);
            }
            if (('ionDecrease' === en)) {
              const pd_2:any = ((<any>_co._keyChg(false,false)) !== false);
              ad = (pd_2 && ad);
            }
            return ad;
          },i1.View_RangeKnob_0,i1.RenderType_RangeKnob)),i0.ɵdid(49152,(null as any),
          0,i2.RangeKnob,([] as any[]),{ratio:[0,'ratio'],pressed:[1,'pressed'],pin:[2,
              'pin'],min:[3,'min'],max:[4,'max'],val:[5,'val'],disabled:[6,'disabled'],
              labelId:[7,'labelId']},{ionIncrease:'ionIncrease',ionDecrease:'ionDecrease'}),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_Range_2)),
      i0.ɵdid(16384,(null as any),0,i3.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),i0.ɵncd((null as any),1)],(_ck,_v) => {
    var _co:i4.Range = _v.component;
    const currVal_0:any = _co._ticks;
    _ck(_v,4,0,currVal_0);
    const currVal_13:any = _co._ratioA;
    const currVal_14:any = _co._pressedA;
    const currVal_15:any = _co._pin;
    const currVal_16:any = _co._min;
    const currVal_17:any = _co._max;
    const currVal_18:any = _co._valA;
    const currVal_19:any = _co._disabled;
    const currVal_20:any = _co._labelId;
    _ck(_v,8,0,currVal_13,currVal_14,currVal_15,currVal_16,currVal_17,currVal_18,currVal_19,
        currVal_20);
    const currVal_21:any = _co._dual;
    _ck(_v,10,0,currVal_21);
  },(_ck,_v) => {
    var _co:i4.Range = _v.component;
    const currVal_1:any = _co._barL;
    const currVal_2:any = _co._barR;
    _ck(_v,6,0,currVal_1,currVal_2);
    const currVal_3:any = i0.ɵnov(_v,8).pressed;
    const currVal_4:any = ((i0.ɵnov(_v,8).val === i0.ɵnov(_v,8).min) || (i0.ɵnov(_v,
        8).val === (undefined as any)));
    const currVal_5:any = (i0.ɵnov(_v,8).val === i0.ɵnov(_v,8).max);
    const currVal_6:any = i0.ɵnov(_v,8)._x;
    const currVal_7:any = i0.ɵnov(_v,8).val;
    const currVal_8:any = i0.ɵnov(_v,8).min;
    const currVal_9:any = i0.ɵnov(_v,8).max;
    const currVal_10:any = i0.ɵnov(_v,8).disabled;
    const currVal_11:any = i0.ɵnov(_v,8).labelId;
    const currVal_12:any = (i0.ɵnov(_v,8).disabled? (0 - 1): 0);
    _ck(_v,7,0,currVal_3,currVal_4,currVal_5,currVal_6,currVal_7,currVal_8,currVal_9,
        currVal_10,currVal_11,currVal_12);
  });
}
export function View_Range_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'ion-range',([] as any[]),
      [[2,'range-disabled',(null as any)],[2,'range-pressed',(null as any)],[2,'range-has-pin',
          (null as any)]],(null as any),(null as any),View_Range_0,RenderType_Range)),
      i0.ɵprd(5120,(null as any),i5.NG_VALUE_ACCESSOR,(p0_0:any) => {
        return [p0_0];
      },[i4.Range]),i0.ɵdid(1228800,(null as any),0,i4.Range,[i6.Form,i7.Haptic,[2,
          i8.Item],i9.Config,i10.Platform,i0.ElementRef,i0.Renderer,i11.DomController,
          i0.ChangeDetectorRef],(null as any),(null as any))],(null as any),(_ck,_v) => {
    const currVal_0:any = i0.ɵnov(_v,2)._disabled;
    const currVal_1:any = i0.ɵnov(_v,2)._pressed;
    const currVal_2:any = i0.ɵnov(_v,2)._pin;
    _ck(_v,0,0,currVal_0,currVal_1,currVal_2);
  });
}
export const RangeNgFactory:i0.ComponentFactory<i4.Range> = i0.ɵccf('ion-range',i4.Range,
    View_Range_Host_0,{color:'color',mode:'mode',disabled:'disabled',min:'min',max:'max',
        step:'step',snaps:'snaps',pin:'pin',debounce:'debounce',dualKnobs:'dualKnobs'},
    {ionFocus:'ionFocus',ionChange:'ionChange',ionBlur:'ionBlur'},['[range-left]',
        '[range-right]']);
