// ionic-angular/components/infinite-scroll/infinite-scroll-content.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '../spinner/spinner.ngfactory';
import * as i2 from 'ionic-angular/components/spinner/spinner';
import * as i3 from 'ionic-angular/config/config';
import * as i4 from '@angular/common';
import * as i5 from 'ionic-angular/components/infinite-scroll/infinite-scroll-content';
import * as i6 from 'ionic-angular/components/infinite-scroll/infinite-scroll';
const styles_InfiniteScrollContent:any[] = ([] as any[]);
export const RenderType_InfiniteScrollContent:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_InfiniteScrollContent,data:{}});
function View_InfiniteScrollContent_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'div',[['class',
      'infinite-loading-spinner']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'ion-spinner',
      ([] as any[]),[[2,'spinner-paused',(null as any)]],(null as any),(null as any),
      i1.View_Spinner_0,i1.RenderType_Spinner)),i0.ɵdid(114688,(null as any),0,i2.Spinner,
      [i3.Config,i0.ElementRef,i0.Renderer],{name:[0,'name']},(null as any))],(_ck,
      _v) => {
    var _co:any = _v.component;
    const currVal_1:any = _co.loadingSpinner;
    _ck(_v,2,0,currVal_1);
  },(_ck,_v) => {
    const currVal_0:any = i0.ɵnov(_v,2)._paused;
    _ck(_v,1,0,currVal_0);
  });
}
function View_InfiniteScrollContent_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),0,'div',[['class',
      'infinite-loading-text']],[[8,'innerHTML',1]],(null as any),(null as any),(null as any),
      (null as any)))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.loadingText;
    _ck(_v,0,0,currVal_0);
  });
}
export function View_InfiniteScrollContent_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),4,'div',[['class',
      'infinite-loading']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),
      View_InfiniteScrollContent_1)),i0.ɵdid(16384,(null as any),0,i4.NgIf,[i0.ViewContainerRef,
      i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵand(16777216,(null as any),
      (null as any),1,(null as any),View_InfiniteScrollContent_2)),i0.ɵdid(16384,(null as any),
      0,i4.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any))],
      (_ck,_v) => {
        var _co:i5.InfiniteScrollContent = _v.component;
        const currVal_0:any = _co.loadingSpinner;
        _ck(_v,2,0,currVal_0);
        const currVal_1:any = _co.loadingText;
        _ck(_v,4,0,currVal_1);
      },(null as any));
}
export function View_InfiniteScrollContent_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'ion-infinite-scroll-content',
      ([] as any[]),[[1,'state',0]],(null as any),(null as any),View_InfiniteScrollContent_0,
      RenderType_InfiniteScrollContent)),i0.ɵdid(114688,(null as any),0,i5.InfiniteScrollContent,
      [i6.InfiniteScroll,i3.Config],(null as any),(null as any))],(_ck,_v) => {
    _ck(_v,1,0);
  },(_ck,_v) => {
    const currVal_0:any = i0.ɵnov(_v,1).inf.state;
    _ck(_v,0,0,currVal_0);
  });
}
export const InfiniteScrollContentNgFactory:i0.ComponentFactory<i5.InfiniteScrollContent> = i0.ɵccf('ion-infinite-scroll-content',
    i5.InfiniteScrollContent,View_InfiniteScrollContent_Host_0,{loadingSpinner:'loadingSpinner',
        loadingText:'loadingText'},{},([] as any[]));
