// ionic-angular/components/refresher/refresher-content.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from 'ionic-angular/components/icon/icon';
import * as i2 from 'ionic-angular/config/config';
import * as i3 from '@angular/common';
import * as i4 from '../spinner/spinner.ngfactory';
import * as i5 from 'ionic-angular/components/spinner/spinner';
import * as i6 from 'ionic-angular/components/refresher/refresher-content';
import * as i7 from 'ionic-angular/components/refresher/refresher';
const styles_RefresherContent:any[] = ([] as any[]);
export const RenderType_RefresherContent:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_RefresherContent,data:{}});
function View_RefresherContent_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'div',[['class',
      'refresher-pulling-icon']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'ion-icon',[['role',
      'img']],[[2,'hide',(null as any)]],(null as any),(null as any),(null as any),
      (null as any))),i0.ɵdid(147456,(null as any),0,i1.Icon,[i2.Config,i0.ElementRef,
      i0.Renderer],{name:[0,'name']},(null as any))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_1:any = _co.pullingIcon;
    _ck(_v,2,0,currVal_1);
  },(_ck,_v) => {
    const currVal_0:any = i0.ɵnov(_v,2)._hidden;
    _ck(_v,1,0,currVal_0);
  });
}
function View_RefresherContent_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),0,'div',[['class',
      'refresher-pulling-text']],[[8,'innerHTML',1]],(null as any),(null as any),(null as any),
      (null as any)))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.pullingText;
    _ck(_v,0,0,currVal_0);
  });
}
function View_RefresherContent_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),0,'div',[['class',
      'refresher-refreshing-text']],[[8,'innerHTML',1]],(null as any),(null as any),
      (null as any),(null as any)))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.refreshingText;
    _ck(_v,0,0,currVal_0);
  });
}
export function View_RefresherContent_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),4,'div',[['class',
      'refresher-pulling']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),
      View_RefresherContent_1)),i0.ɵdid(16384,(null as any),0,i3.NgIf,[i0.ViewContainerRef,
      i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵand(16777216,(null as any),
      (null as any),1,(null as any),View_RefresherContent_2)),i0.ɵdid(16384,(null as any),
      0,i3.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),5,'div',[['class','refresher-refreshing']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),2,'div',[['class','refresher-refreshing-icon']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),1,'ion-spinner',([] as any[]),
          [[2,'spinner-paused',(null as any)]],(null as any),(null as any),i4.View_Spinner_0,
          i4.RenderType_Spinner)),i0.ɵdid(114688,(null as any),0,i5.Spinner,[i2.Config,
          i0.ElementRef,i0.Renderer],{name:[0,'name']},(null as any)),(_l()(),i0.ɵand(16777216,
          (null as any),(null as any),1,(null as any),View_RefresherContent_3)),i0.ɵdid(16384,
          (null as any),0,i3.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},
          (null as any))],(_ck,_v) => {
    var _co:i6.RefresherContent = _v.component;
    const currVal_0:any = _co.pullingIcon;
    _ck(_v,2,0,currVal_0);
    const currVal_1:any = _co.pullingText;
    _ck(_v,4,0,currVal_1);
    const currVal_3:any = _co.refreshingSpinner;
    _ck(_v,8,0,currVal_3);
    const currVal_4:any = _co.refreshingText;
    _ck(_v,10,0,currVal_4);
  },(_ck,_v) => {
    const currVal_2:any = i0.ɵnov(_v,8)._paused;
    _ck(_v,7,0,currVal_2);
  });
}
export function View_RefresherContent_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'ion-refresher-content',
      ([] as any[]),[[1,'state',0]],(null as any),(null as any),View_RefresherContent_0,
      RenderType_RefresherContent)),i0.ɵdid(114688,(null as any),0,i6.RefresherContent,
      [i7.Refresher,i2.Config],(null as any),(null as any))],(_ck,_v) => {
    _ck(_v,1,0);
  },(_ck,_v) => {
    const currVal_0:any = i0.ɵnov(_v,1).r.state;
    _ck(_v,0,0,currVal_0);
  });
}
export const RefresherContentNgFactory:i0.ComponentFactory<i6.RefresherContent> = i0.ɵccf('ion-refresher-content',
    i6.RefresherContent,View_RefresherContent_Host_0,{pullingIcon:'pullingIcon',pullingText:'pullingText',
        refreshingSpinner:'refreshingSpinner',refreshingText:'refreshingText'},{},
    ([] as any[]));
